ports/esp32/Makefile

```diff
@@ -382,6 +382,8 @@ OBJ_MP += $(addprefix $(BUILD)/, $(EXTMOD_SRC_C:.c=.o))
 OBJ_MP += $(addprefix $(BUILD)/, $(LIB_SRC_C:.c=.o))
 OBJ_MP += $(addprefix $(BUILD)/, $(DRIVERS_SRC_C:.c=.o))
 
+$(OBJ_MP): CFLAGS += -Wdouble-promotion -Wfloat-conversion
```

Comment:
Does this need to be restricted to OBJ_MP, or can it be applied to global CFLAGS?

Reply:
Doubt it: I made separate flags wherever it wasn't doable to apply the flags to the whole source. Don't remember what the problem was here, either libm or the esp-idf SDK. Will check again and add comments.